lib/Mojo/Util.pm

```diff
@@ -88,6 +88,33 @@ monkey_patch(__PACKAGE__, 'sha1_sum',      \&sha1_hex);
 monkey_patch(__PACKAGE__, 'steady_time',
   MONOTONIC ? sub () { Time::HiRes::clock_gettime(Time::HiRes::CLOCK_MONOTONIC()) } : \&Time::HiRes::time);
 
+# Building random bytes function
+monkey_patch(__PACKAGE__, 'urandom_bytes', $^O eq 'MSWin32' ? _build_urandom_bytes_win32() : _build_urandom_bytes());
+
+sub _build_urandom_bytes_win32 {
+  require Win32::API;
+
+  my $RtlGenRandom = Win32::API->new('advapi32', 'INT SystemFunction036 (PVOID RandomBuffer, ULONG RandomBufferLength)')
+    or croak "SystemFunction036 import failed: $!";
```

Comment:
Is this actually used in our Windows CI tests? This seems like code nobody on the team can maintain long term and we will end up removing it because of that the first it causes problems.

Reply:
Tests are running `SystemFunction036` (RtlGenRandom) on Windows to get random bytes. It's used in the same way as for example [Crypt::URandom](https://metacpan.org/release/DDICK/Crypt-URandom-0.40/source/lib/Crypt/URandom.pm#L92-97). However, It is a legacy API, so we can consider using `ProcessPrng` instead.
   
Here are some discussions in rust and golang:
   - https://github.com/rust-random/rand/issues/111
   - https://github.com/golang/go/issues/53192

Reply:
This part of the code will probably end up being a blocker, since i don't see anyone on the team agreeing to take responsibility for maintaining it in the future.

Reply:
I'd not expect finding such code in web framework. In the same time, AFAIK the alternatives are:
- using some small old library with single maintainer and with no recent releases, that does exactly this call, but conveniently wrapped, 
- overkill of making Net-SSLeay a dependency to get it through openssl binding.

Reply:
`Net::SSLeay` is already a soft dependency, maybe there's a clean way to make this feature optional? A hard dependency is pretty much out of the question. It's unfortunate Perl doesn't ship with some kind of OpenSSL binding, for mojo.js we've chosen a different solution and [just encrypt all sessions](https://github.com/mojolicious/mojo.js/blob/main/src/session.ts).

Reply:
`Crypt::URandom` has a pile of different window support calls. Adding that as an optional dependency for windows enjoyers might unblock this.

Reply:
`Crypt::URandom` appears to [sets up some context](https://metacpan.org/release/DDICK/Crypt-URandom-0.40/source/lib/Crypt/URandom.pm#L63-88) for `CryptGenRandom` for Win2k, and uses `RtlGenRandom` for later Windows versions.

According to a discussion in golang crypto/rand, it seems fine to use `RtlGenRandom` (SystemFunction036) https://github.com/golang/go/issues/33542#issuecomment-715642267

Reply:
What about something like this?

```perl

use constant CRYPT_URANDOM => !!eval { require Crypt::URandom };

...

sub urandom_bytes {
  my $num   = shift || 32;

  return Crypt::URandom::urandom($num) if CRYPT_URANDOM;

  croak 'Cannot find /dev/urandom. On Windows? Try installing "Crypt::URandom" from CPAN' 
    unless -f '/dev/urandom';

  open(my $urandom, '<', '/dev/urandom')  or croak "Cannot open /dev/urandom: $!";
  sysread($urandom, $bytes, $num) == $num or die "sysread() from /dev/urandom didn't return $num bytes";
  close($urandom);

  return $bytes;
}

```

Reply:
I've updated the PR with logic similar to the above, avoiding any Win32 specific code.

Also added `Crypt::URandom` to the Windows GH actions workflow, should the be in Makefile.PL instead?